server/src/main/java/org/opensearch/index/mapper/ContentPath.java

```diff
@@ -84,6 +84,9 @@ public void remove() {
     }
 
     public String pathAsText(String name) {
+        if (offset == index) {
+            return name;
+        }
         sb.setLength(0);
```

Comment:
Do `sb` need to be a class member? It looks like `sb` is only used by this method and its state is reset every time the method is invoked. This method could become:

```java
if (offset == index) {
    return name;
}
final StringBuilder sb = new StringBuilder();
for (int i = offset; i < index; i++) {
    sb.append(path[i]).append(DELIMITER);
}
sb.append(name);
return sb.toString();

```

Reply:
I guess the existing code was written like this so as to not allocate any additional StringBuilder instances since pathAsText may be called multiple times on a single ContentPath instance, and setLength(0) will be faster than new allocations in such cases.